### PR TITLE
Video page: add song title + remove song license on music videos

### PR DESCRIPTION
**Pull request by ChunkyProgrammer**
adds the song title to the track
Includes musicTracks in the api

Test Videos:
https://www.youtube.com/watch?v=M3e5R48f4Ks
https://www.youtube.com/watch?v=QJJYpsA5tv8
https://www.youtube.com/watch?v=KQsrIxyoJdE

image
Closes https://github.com/iv-org/invidious/issues/3670

### CODE REVIEW

File: src/invidious/videos/parser.cr [hunk 1]
```diff
@@ -325,17 +325,27 @@ def parse_video_info(video_id : String, player_response : Hash(String, JSON::Any
     album = nil
     music_license = nil
 
+    # used when multiple songs
+    song = music_desc.dig?("carouselLockupRenderer", "videoLockup", "compactVideoRenderer", "title", "simpleText")
+
+    # used when multiple songs and the song has a link
+    if !song
+      song = music_desc.dig("carouselLockupRenderer", "videoLockup", "compactVideoRenderer", "title", "runs", 0, "text")
+    end
```
Comment (SamantazFox) [inline src/invidious/videos/parser.cr:334]:
```suggestion
    # Used when the video has multiple songs
    if song_title = music_desc.dig?("carouselLockupRenderer", "videoLockup", "compactVideoRenderer", "title")
      # "simpleText" for plain text / "runs" when song has a link
      song = song_title["simpleText"]? || song_title.dig("runs", 0, "text")
    end
```

**Review by SamantazFox — APPROVED** [review]
Code looks good to me! I've proposed some style changes.

**Comment by SamantazFox** [comment]
Deployed on https://test.invidious.io.
It's working perfectly ^^

I'm wondering if we should remove the license infos, though.
It feels like user clutter:
image

**Comment by ChunkyProgrammer** [comment]
Would you like me to remove the license from the ecr file?

**Comment by SamantazFox** [comment]
> Would you like me to remove the license from the ecr file?

Yes, please :)

**Comment by ChunkyProgrammer** [comment]
@SamantazFox Updated! 🙂

**Review by SamantazFox — APPROVED** [review]

**Comment by SamantazFox** [comment]
Thanks for your contribution to invidious :D

**Pull request merged**